src/lib/y2storage/proposal/autoinst_devices_planner.rb

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/disk"
+
+module Y2Storage
+  module Proposal
+    # Class to generate a list of Planned::Device objects that must be allocated
+    # during the AutoYaST proposal.
+    class AutoinstDevicesPlanner
+      include Yast::Logger
+
+      # Constructor
+      #
+      # @param devicegraph [Devicegraph] Devicegraph to be used as starting point
+      def initialize(devicegraph)
+        @devicegraph = devicegraph
+      end
+
+      # Returns an array of planned devices according to the drives map
+      #
+      # @raise [Error] if not partitions were specified
+      #
+      # @param drives_map [Proposal::AutoinstDrivesMap] Drives map from AutoYaST
+      # @return [Array<Planned::Partition>] List of planned partitions
+      def planned_devices(drives_map)
+        result = []
+
+        drives_map.each_pair do |disk_name, drive_spec|
+          disk = Disk.find_by_name(devicegraph, disk_name)
+          result.concat(planned_for_disk(disk, drive_spec))
+        end
+
+        if result.empty?
+          raise Error, "No partitions specified"
+        end
+
+        checker = BootRequirementsChecker.new(devicegraph, planned_devices: result)
+        result.concat(checker.needed_partitions)
+
+        result
+      end
+
+    protected
+
+      # @return [Devicegraph] Starting devicegraph
+      attr_reader :devicegraph
+
+      # Returns an array of planned partitions for a given disk
+      #
+      # @param disk         [Disk] Disk to place the partitions on
+      # @param partitioning [Hash] Partitioning specification from AutoYaST
+      # @return [Array<Planned::Partition>] List of planned partitions
+      def planned_for_disk(disk, spec)
+        result = []
+        spec.fetch("partitions", []).each do |part_description|
+          # TODO: fix Planned::Partition.initialize
+          part = Y2Storage::Planned::Partition.new(nil, nil)
+          part.disk = disk.name
+          # part.bootable no está en el perfil (¿existe lógica?)
```

Comment:
I am looking forward to learn some pepper offensive words in spanish, but google translate dissatisfied me with result 😉